dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotcms.repackage.org.apache.commons.lang.StringUtils;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Logger;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This task updates the definition of the {@code workflow_action} table, creates the new
+ * intermediate {@code workflow_action_step} table, and updates the definition of a table index:
+ * <ul>
+ * <li>A new column called {@code scheme_id} will be added to the {@code workflow_action} table.
+ * This allows actions to be assigned to Workflow Schemes and not to steps directly.</li>
+ * <li>The new {@code workflow_action_step} table allows an action to be associated to one or more
+ * steps (an N-N relationship).</li>
+ * <li>The index created for the {@code workflow_scheme_x_structure} column will be updated to NOT
+ * be unique.</li>
+ * </ul>
+ *
+ * @author Jose Castro
+ * @version 4.3.0
+ * @since Nov 1st, 2017
+ */
+public class Task04305UpdateWorkflowActionTable implements StartupTask {
+
+    private static final String MYSQL_FIND_INTERMEDIATE_TABLE    = "SELECT * from workflow_action_step";
+    private static final String POSTGRES_FIND_INTERMEDIATE_TABLE = "SELECT * from workflow_action_step";
+    private static final String MSSQL_FIND_INTERMEDIATE_TABLE    = "SELECT * from workflow_action_step";
+    private static final String ORACLE_FIND_INTERMEDIATE_TABLE   = "SELECT * from workflow_action_step";
+
+    private static final String MYSQL_CREATE_INTERMEDIATE_TABLE = "CREATE TABLE workflow_action_step (action_id VARCHAR(36) NOT NULL, step_id VARCHAR(36) NOT NULL, action_order INT default 0)";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 193 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)